clang/lib/CodeGen/CodeGenPGO.cpp

```diff
@@ -959,8 +959,12 @@ void CodeGenPGO::emitCounterIncrement(CGBuilderTy &Builder, const Stmt *S,
 
   unsigned Counter = (*RegionCounterMap)[S];
 
-  llvm::Value *Args[] = {FuncNameVar,
-                         Builder.getInt64(FunctionHash),
+  // Make sure that pointer to global is passed in with zero addrspace
+  // This is relevant during GPU profiling
+  auto *NormalizedPtr = llvm::ConstantExpr::getPointerBitCastOrAddrSpaceCast(
+      FuncNameVar, llvm::PointerType::getUnqual(CGM.getLLVMContext()));
```

Comment:
Avoid using getUnqual. I'm sure there's a proper address space to query somewhere here

Reply:
`FuncNameVar` has an addrspace of 1 on GPU devices. The `instrprof_increment` throws a type error if you pass in a pointer without addrspace 0. Is there a reason why one should use `PointerType::get(ctx, 0)` over `PointerType::getUnqual`? Is the latter bad practice/being deprecated?

Reply:
@arsenm I think this use is a generic void*, I expect we need/want AS 0 here, always.

Reply:
But generic is a language concept, which doesn't necessarily map to IR AS0. Technically you should go through the language addrspace map

Reply:
No, because the runtime, which defines the functions called here, does neither. It expects AS0 calls, which is what we should be generating. If the runtime was "language aware", this might be different.